cmd/apichecker/apicheck.go

```diff
@@ -0,0 +1,1344 @@
+					Key:           tempData.Key,
+					RegExp:        tempData.RegExp,
+					TextTokenData: tempData.TextTokenData,
+					TokenData:     tempData.TokenData,
+					TokenDataEnd:  tempData.TokenDataEnd,
+					Val:           tempData.Val,
+					Path:          tempData.Path},
+			},
+		}, nil
+	default:
+		return ExchangeInfo{}, errors.New("invalid checkType")
+	}
+}
+
+// HTMLScrapeDefault gets check string data for the default cases
+func HTMLScrapeDefault(htmlData HTMLScrapingData) ([]string, error) {
+	var resp []string
+	temp, err := http.Get(htmlData.Path)
+	if err != nil {
+		return resp, err
+	}
+	tokenizer := html.NewTokenizer(temp.Body)
+loop:
+	for {
+		next := tokenizer.Next()
+		switch next {
+		case html.ErrorToken:
+			break loop
+		case html.StartTagToken:
+			token := tokenizer.Token()
+			if token.Data == htmlData.TokenData {
+				for _, a := range token.Attr {
+					if a.Key == htmlData.Key && a.Val == htmlData.Val {
+					loop2:
+						for {
+							nextToken := tokenizer.Next()
+							switch nextToken {
+							case html.EndTagToken:
+								t := tokenizer.Token()
+								if t.Data == htmlData.TokenDataEnd {
+									break loop2
+								}
+							case html.StartTagToken:
+								newtok := tokenizer.Token()
+								if newtok.Data == htmlData.TextTokenData {
+									inner := tokenizer.Next()
+									if inner == html.TextToken {
+										tempStr := string(tokenizer.Text())
+										r, err := regexp.Compile(htmlData.RegExp)
+										if err != nil {
+											return resp, err
+										}
+										result := r.MatchString(tempStr)
+										if result {
+											appendStr := r.FindString(tempStr)
+											resp = append(resp, appendStr)
+										}
+									}
+								}
+							}
+						}
+					}
+				}
+			}
+		default:
+			continue
+		}
+	}
+	return resp, nil
+}
+
+// HTMLScrapeBTSE gets the check string for BTSE exchange
+func HTMLScrapeBTSE(htmlData HTMLScrapingData) ([]string, error) {
+	var resp []string
+	temp, err := http.Get(htmlData.Path)
+	if err != nil {
+		return resp, err
+	}
+	tokenizer := html.NewTokenizer(temp.Body)
+loop:
+	for {
+		next := tokenizer.Next()
+		switch next {
+		case html.ErrorToken:
+			break loop
+		case html.StartTagToken:
+			token := tokenizer.Token()
+			if token.Data == htmlData.TokenData {
+				for _, z := range token.Attr {
+					if z.Key == htmlData.Key && z.Val == htmlData.Val {
+						inner := tokenizer.Next()
+						if inner == html.TextToken {
+							resp = append(resp, string(tokenizer.Text()))
+						}
+					}
+				}
+			}
+		}
+	}
+	return resp, nil
+}
+
+// HTMLScrapeBitfinex gets the check string for Bitfinex exchange
+func HTMLScrapeBitfinex(htmlData HTMLScrapingData) ([]string, error) {
+	var resp []string
+	temp, err := http.Get(htmlData.Path)
+	if err != nil {
+		return resp, err
+	}
+	tokenizer := html.NewTokenizer(temp.Body)
+loop:
+	for {
+		next := tokenizer.Next()
+		switch next {
+		case html.ErrorToken:
+			break loop
```

Comment:
This looks like a common thing you've done, so I'll stop commenting on all of them. If you are only trying to go to the next iteration of the innermost loop you are in, just use `continue`